subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformExecutionBuildOperationIntegrationTest.groovy

```diff
@@ -176,6 +205,371 @@ class ArtifactTransformExecutionBuildOperationIntegrationTest extends AbstractIn
+
+                    artifactTypes {
+                        jar {
+                            attributes.attribute(color, 'blue')
+                        }
+                    }
+                }
+            }
+        """
+
+        when:
+        withBuildCache().fails(":consumer:resolve")
+        then:
+        def projectExecution = findTransformExecution { it.artifactName == 'producer.jar' }
+        projectExecution.failure == "java.lang.RuntimeException: BOOM"
+        with(projectExecution.result) {
+            skipMessage == null
+            originExecutionTime == null
+            originBuildInvocationId == null
+            executionReasons == ['No history is available.']
+            cachingDisabledReasonMessage == null
+            cachingDisabledReasonCategory == null
+        }
+        with(snapshotInputsOperation(projectExecution).result) {
+            hash != null
+            classLoaderHash != null
+            implementationClassName == 'MakeGreen'
+            inputValueHashes.keySet() ==~ ['inputArtifactPath', 'inputPropertiesHash']
+            outputPropertyNames == ['outputDirectory', 'resultsFile']
+            inputFileProperties.keySet() ==~ ['inputArtifact', 'inputArtifactDependencies']
+            with(inputFileProperties.inputArtifact) {
+                hash != null
+                attributes == ['DIRECTORY_SENSITIVITY_DEFAULT', 'FINGERPRINTING_STRATEGY_RELATIVE_PATH', 'LINE_ENDING_SENSITIVITY_NORMALIZE_LINE_ENDINGS']
+                roots
+            }
+        }
+        actionExecution(projectExecution) != null
+    }
+
+    def "classpath notation #classpathNotation transforms can be captured"() {
+        settingsFile << """
+            include 'producer', 'consumer'
+        """
+
+        setupBuildWithColorTransform()
+        buildFile << """
+            abstract class MakeGreen implements TransformAction<TransformParameters.None> {
+                @InputArtifact
+                abstract Provider<FileSystemLocation> getInputArtifact()
+
+                void transform(TransformOutputs outputs) {
+                    def input = inputArtifact.get().asFile
+                    println "processing [\${input.name}]"
+                    assert input.file
+                    def output = outputs.file(input.name + ".green")
+                    output.text = input.name + ".green"
+                }
+            }
+        """
+
+        // We need to add a Kotlin build file here so we have access to `gradleKotlinDsl()`
+        file("consumer/build.gradle.kts") << """
+            val color = Attribute.of("color", String::class.java)
+            dependencies {
+                implementation($classpathNotation)
+
+                artifactTypes {
+                    create("jar") {
+                        attributes.attribute(color, "blue")
+                    }
+                }
+            }
+        """
+
+        when:
+        withBuildCache().run ":consumer:resolve"
```

Comment:
❔  Should we add another when/then with a consecutive `withBuildCache` invocation, or should we drop it here?

There are more tests below that also only do one invocation with build cache

Reply:
I use the `withBuildCache()`, so Gradle fills out the snapshotting operation details. If you don't do that, Gradle will not capture a build cache key or the input/output hashes. Maybe it is time again to try to switch this on every time on the Gradle side. We didn't do this since it has been causing small performance regressions. The code where we currently decide this is [here](https://github.com/gradle/gradle/blob/b50d04bf77db892c8186f21f40cdd8e468ce4c25/subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java#L115).

Reply:
Maybe it's worth a comment in the source code, then?

Also, IIUC this test does not actually assert against those snapshot operations, compared to other tests like `"cacheability information for artifact transform executions is captured"`. So, maybe we can still drop the `withBuildCache()` from some of these tests? 🤔